File: Assets/Plugins/iOS/PlayPhoneMultiNet/unity/providers/MNClientRobotsProviderUnity.m
//
//  MNClientRobotsProviderUnity.m
//  PlayPhone SDK for Unity
//
//  Copyright (c) 2012 PlayPhone Inc. All rights reserved.
//

#import "MNDirect.h"
#import "MNClientRobotsProvider.h"

#import "MNClientRobotsProviderUnity.h"

EXTERN_C
bool _MNClientRobotsProvider_IsRobot(const char* userInfo) {
    MARK;

    return [[MNDirect clientRobotsProvider] isRobot:[[MNUnity serializer] deserialize:[MNUserInfo class] fromJson:[NSString stringWithUTF8String:userInfo]]];
}

EXTERN_C
void _MNClientRobotsProvider_PostRobotScore(const char* userInfo, long long score) {
    MARK;

    [[MNDirect clientRobotsProvider] postRobot:[[MNUnity serializer] deserialize:[MNUserInfo class] fromJson:[NSString stringWithUTF8String:userInfo]] score:score];
}

EXTERN_C
void _MNClientRobotsProvider_SetRoomRobotLimit(int robotCount) {
    MARK;

    [[MNDirect clientRobotsProvider] setRoomRobotLimit:robotCount];
}

EXTERN_C
int _MNClientRobotsProvider_GetRoomRobotLimit() {
    MARK;

    return [[MNDirect clientRobotsProvider] getRoomRobotLimit];
}

File: Assets/Plugins/Android/src/com/playphone/multinet/unity/MNClientRobotsProviderUnity.java
package com.playphone.multinet.unity;

import com.playphone.multinet.*;
import com.unity3d.player.UnityPlayer;

public class MNClientRobotsProviderUnity {
    public static void shutdown() {
        MNUnity.MARK();

        UnityPlayer.currentActivity.runOnUiThread(new Runnable()
        {
            public void run()
            {
                MNDirect.getClientRobotsProvider().shutdown();
            }
        });
    }

    public static boolean isRobot(final String userInfo) {
        MNUnity.MARK();

        return MNDirect.getClientRobotsProvider().isRobot(MNUnity.serializer.deserialize(userInfo,MNUserInfo.class));
    }

    public static void postRobotScore(final String userInfo, final long score) {
        MNUnity.MARK();

        UnityPlayer.currentActivity.runOnUiThread(new Runnable()
        {
            public void run()
            {
                MNDirect.getClientRobotsProvider().postRobotScore(MNUnity.serializer.deserialize(userInfo,MNUserInfo.class), score);
            }
        });
    }

    public static void setRoomRobotLimit(final int robotCount) {
        MNUnity.MARK();

        UnityPlayer.currentActivity.runOnUiThread(new Runnable()
        {
            public void run()
            {
                MNDirect.getClientRobotsProvider().setRoomRobotLimit(robotCount);
            }
        });
    }

    public static int getRoomRobotLimit() {
        MNUnity.MARK();

        return MNDirect.getClientRobotsProvider().getRoomRobotLimit();
    }

}

